src/cli/cmds/generate.js

```diff
@@ -0,0 +1,88 @@
+import Generate from '../../sub-commands/generate';
+import Blueprint from '../../models/blueprint';
+
+const subCommand = new Generate();
+
+const usage = `Usage:
+  $0 generate <blueprint> <name> [options]
+  $0 help generate <blueprint>`;
+
+module.exports = {
+  command: 'generate <blueprint> <name>',
+  aliases: ['g', 'gen'],
+  describe: 'Generate project file(s) from a blueprint',
+  builder: yargs => {
+    yargs
+      .usage(usage)
+      .option('dry-run', {
+        alias: 'd',
+        describe: "List files but don't generate them", // eslint-disable-line
+        type: 'boolean'
+      })
+      .option('verbose', {
+        alias: 'v',
+        describe: 'Verbose output, including file contents',
+        type: 'boolean'
+      })
+      .group(['dry-run', 'verbose', 'help'], 'Generate Options:')
+      .updateStrings({
+        'Commands:': 'Blueprints:',
+        'Options:': 'Blueprint Options:'
+      });
+    buildBlueprintCommands(yargs);
+  },
+  handler: argv => console.log(`Unrecognised blueprint '${argv.blueprint}'`)
+};
+
+const buildBlueprintCommands = yargs => {
+  const customCommand = blueprint => {
+    let custom = blueprint.command || {};
+    let builder = custom.builder || {};
+
+    if (typeof builder === 'function' && builder.length === 2) {
+      const wrappedBuilder = builder;
+      builder = yargs => wrappedBuilder(yargs, blueprint);
+      // opportunity here to interrogate yargs for configuration
+    }
+
+    // could do more to understand if options have been specified
```

Comment:
This is my working theory for `blueprintrc`  

```json
{
   "paths": {
    "smart": "containers",
    "dumb": "components",
    "blueprint": "~/.blueprints",
    "sourceBase": "sd",
    "testBase": "a",
    "fileCasing": "snake"
  },
  "blueprintPaths": [
    "/home/user/workspace/project/blueprints",
    "/home/user/.local/blueprints",
    "@someNppmNotSureAboutThis",
    "@butNeedSomeWay",
    "@toTellPathAndModule",
    "@apart"
  ],
  "defaults": {
    "git": {
      "add": true
    }
  },
  "bp.store": {
    "sagas": false,
    "selectors": {
      "separateFile": "selectors.js"
    }
  },
  "bp.blueprint": {
    "command": true
  }
}
```

Reply:
or `{"bp": {"blueprint": {"command": true}}}`

Reply:
I touched on this earlier. I quite like format of:

```
options: {
  default: {
    propTypes: true
  },
  blueprint: {
    command: true
  }
}
```

And having the blueprint loader resolve the options for specific blueprints to keep them simple in the scope of an individual blueprint. It would be fairly simple, one time, logic.

Reply:
:+1:

Reply:
👍

Reply:
Considering discussion on #101 how about:

```
bp: {
  common: {
    propTypes: true
  },
  blueprint: {
    command: true
  }
}